app/src/main/java/de/tum/in/tumcampusapp/widgets/MVVWidget.java

```diff
@@ -121,33 +125,36 @@ static void updateAppWidgets(Context context, AppWidgetManager appWidgetManager,
      * @param appWidgetId the id of the widget to update
      */
     static void updateAppWidget(Context context, AppWidgetManager appWidgetManager,
-                                int appWidgetId) {
-
+                                int appWidgetId, boolean forceLoadData) {
         // Get the settings for this widget from the database
-        TransportManager transportManager = new TransportManager(context);
-        TransportManager.WidgetDepartures widgetDepartures = transportManager.getWidget(appWidgetId);
+        WidgetDepartures widgetDepartures = transportManager.getWidget(appWidgetId);
 
         // Instantiate the RemoteViews object for the app widget layout.
         RemoteViews rv = new RemoteViews(context.getPackageName(), R.layout.mvv_widget);
-        String station = widgetDepartures.getStation();
-        if (station == null) {
-            station = context.getString(R.string.mvv_widget_no_station);
-        }
-        rv.setTextViewText(R.id.mvv_widget_station, station);
+        rv.setTextViewText(R.id.mvv_widget_station, widgetDepartures.getStation());
 
         // Set up offline symbol (may be shown one update delayed)
         rv.setViewVisibility(R.id.mvv_widget_offline, widgetDepartures.isOffline() ? View.VISIBLE : View.INVISIBLE);
 
         // Set up the configuration activity listeners
         Intent configIntent = new Intent(context, MVVWidgetConfigureActivity.class);
         configIntent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
-        PendingIntent pendingIntent = PendingIntent.getActivity(context, appWidgetId, configIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, appWidgetId * 10, configIntent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
well actually this is an id for the intent, so it does not work, like expected, if there are some with the same id
the most simpel solution was to append an intend index to the widget id:
appWidgetID * 10       -> xxx0
appWidgetID * 10 + 1 -> xxx1